content/posts/2023-05-15-mac-learning.md

```diff
@@ -0,0 +1,303 @@
+root@bridge-a:~# 
+
+Bridge-B:
+root@bridge-b:~# bridge fdb show dynamic
+02:11:11:11:00:00 dev eth0 master br0   <====== NEW (H1)
+02:aa:aa:aa:00:02 dev eth0 master br0 
+02:33:33:33:00:00 dev eth1 master br0 
+02:44:44:44:00:00 dev eth2 master br0 
+root@bridge-b:~# 
+```
+The only thing that differs here is that Bridge-B has learned about H1
+on its port eth0. It learns this from the **broadcast** `ARP Who Has
+10.0.1.2?` sent by H1 (preceding the ICMP Echo Request of the Ping);
+this broadcast ARP is flooded throughout the whole LAN. The 'ARP
+Response' by H2 is sent by **unicast** and never reaches Bridge-B. (More
+precisely, as Bridge-A knows where MAC 02:11:11:11:00:00 resides, it
+forwards the 'ARP Response' directly on eth0.
+
+Now, let H1 send a broadcast ping (`ping 10.0.1.255`). All hosts
+respond.  If we again inspect the MAC databases at the bridges, we see
+they have learned the location of the MACs for all hosts.
+
+```
+Bridge-A:
+root@bridge-a:~# bridge fdb show dynamic
+02:11:11:11:00:00 dev eth0 master br0 
+02:22:22:22:00:00 dev eth1 master br0 
+02:33:33:33:00:00 dev eth2 master br0    <=== NEW (H3)
+02:44:44:44:00:00 dev eth2 master br0    <=== NEW (H4)
+02:bb:bb:bb:00:00 dev eth2 master br0
+root@bridge-a:~# 
+
+Bridge-B:
+root@bridge-b:~# bridge fdb show dynamic
+02:22:22:22:00:00 dev eth0 master br0    <=== NEW (H2)
+02:11:11:11:00:00 dev eth0 master br0 
+02:aa:aa:aa:00:02 dev eth0 master br0 
+02:33:33:33:00:00 dev eth1 master br0 
+02:44:44:44:00:00 dev eth2 master br0 
+root@bridge-b:~# 
+```
+
+The figure below illustrates the MAC databases (FDBs) by using
+symbolic names for the MAC addresses.
+```
+   Bridge-A BaseMAC               Bridge-B BaseMAC
+   02:aa:aa:aa:00:00              02:bb:bb:bb:00:00
+   .----------------------.       .-----------------------.
+   |        Bridge-A      |       |       Bridge-B        |
+   | eth0    eth1    eth2 |       | eth0    eth2     eth2 |
+   '--+-------+-------+---'       '--+-------+--------+---'
+    h1|     h2|   br-b|              |br-a   |h3      |h4    <== FDB
+      |       |     h3'--------------'h1     |        | 
+	  |     .-+--.	h4                h2  .--+-.      |
+	  |     | H2 |	                      | H3 |      |
+	  |     '----'	                      '----'      |
+   .--+-.  IP: 10.0.1.2/24          IP: 10.0.1.3/24 .-+--.
+   | H1 | MAC: 02:22:22:00:00  MAC: 02:33:33::00:00 | H4 |
+   '----'                                           '----'
+ IP: 10.0.1.1/24                            IP: 10.0.1.1/24
+MAC: 02:11:11:11:00:00               MAC: 02:44:44:44::00:00
+```
+
+## Exploring MAC learning with Infix
+
+The examples above were done using a virtual setup with [Infix][Infix] on
+[GNS3][GNS3]. Below are some hints if you like to explore this
+yourself.
+
+To install Infix and GNS3, look at the guide in the  [Infix][Infix]
+README. It is also recommended to look at the [Basic Networking
+Training Post][Training-Basic-Net-Post].
```

Comment:
Maybe introduce _Infix Classic_ in the first paragraph? E.g., "... virtual setup with Infix, specifically the Classic build, on ..."